kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java

```diff
@@ -135,6 +141,12 @@ private String getAdvertisedListener(InetSocketAddress internalListenerAddress,
             log.error("No node for broker {} under loadBalance", internalListenerAddress);
             return null;
         }
+        if (serviceLookupData.get().getProtocol(KafkaProtocolHandler.PROTOCOL_NAME).isPresent()
+                && serviceLookupData.get().getProtocol(KafkaProtocolHandler.PROTOCOL_NAME).get()
+                        .equals(selfAdvertisedListeners)){
```

Comment:
this may be a list, I am not sure we should compare it this way

Reply:
It's the return value of `KafkaProtocolHandler.getProtocolDataToAdvertise()`, which value is `kafkaConfig.getKafkaAdvertisedListeners()`, I compare here to check if the look up result is the broker itself, in this case we cache the look up result since the topic is owned by this broker.

Reply:
makes sense